storage: fix ranges-per-gb balancing

petermattis opened the pull request:
Remove the +/- 0.5 factor from rebalance{From,To}ConvergesOnMean. This
made sense with integer weights but doesn't for the ranges-per-gb
weight. Remove the use of ceil/floor for calculating the
overfull/underfull threshold. Again, this made sense for integer
weights, but not the floating point ranges-per-gb weight.

Fixes #16742

petermattis committed:
storage: fix ranges-per-gb balancing

Remove the +/- 0.5 factor from rebalance{From,To}ConvergesOnMean. This
made sense with integer weights but doesn't for the ranges-per-gb
weight. Remove the use of ceil/floor for calculating the
overfull/underfull threshold. Again, this made sense for integer
weights, but not the floating point ranges-per-gb weight.

Fixes #16742

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/16747)
<!-- Reviewable:end -->

petermattis commented:
I'm investigating the `TestStoreRangeCorruptionChangeReplicas` failure caused by this change.

petermattis commented:
Fixed the test failure. The test was not expecting rebalancing to occur so I added a testing knob which allows rebalancing to be disabled while leaving the replicate queue enabled.

a-robinson reviewed (approved):
LGTM. Sorry for the trouble.
